lib/ansible/module_utils/redfish_utils.py

```diff
@@ -317,15 +313,24 @@ def get_storage_controller_inventory(self):
         result["entries"] = controller_results
         return result
 
-    def get_disk_inventory(self):
+    def get_multi_storage_controller_inventory(self):
+        ret = True
+        entries = []
+        for systems_uri in self.systems_uris:
+            inventory = self.get_storage_controller_inventory(systems_uri)
+            ret = inventory.pop('ret') and ret
+            entries.append(inventory)
+        return dict(ret=ret, entries=entries)
```

Comment:
That `get_multi_something` pattern repeats itself quite a few times in the file, which makes it a good candidate for a refactor into more general function. Something similar to this should work:

    def aggregate(self, func):
        ret = True
        entries = []
        for systems_uri in self.systems_uris:
            inventory = func(systems_uri)
            ret = inventory.pop('ret') and ret
            entries.append(inventory)
        return dict(ret=ret, entries=entries)

    def get_multi_storage_controller_inventory(self):
        return self.aggregate(self.get_storage_controller_inventory)

    def get_multi_disk_inventory(self):
        return self.aggregate(self.get_disk_inventory)

This should reduce the code duplication significantly.

Reply:
Agree this would be a very good improvement.

Reply:
Done :)